packages/app-desktop/gui/ConfigScreen/controls/plugins/FilterForPlugins/FilterForPlugins.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { themeStyle } from '@joplin/lib/theme';
+import uuid from '@joplin/lib/uuid';
+
+interface FilterProps {
+    themeId: number;
+    setFilterValue: Function;
+	onSearchButtonClick: Function;
+	filterValue: string;
+}
+
+export default function FilterForPlugins(props: FilterProps) {
+
+	const theme = themeStyle(props.themeId);
+
+	const filterOptions = ['Select plugin category','Most Downloaded','Recommended', 'Newest', 'Built-in', 'All', 'line-separator', 'Appearance', 'Developer tools', 'Productivity', 'Themes', 'Integrations', 'Viewer', 'Search', 'Tags', 'Editor', 'Files', 'Personal knowledge management', 'line-separator', 'Installed', 'Enabled', 'Disabled', 'Outdated'];
+
+	const clearFilterValue = () => {
```

Comment:
That's a callback - call it something like "onClearFilterValue" and use `useCallback`. Don't forget the set the dependencies correctly which in this case should be props.onSearchButtonClick. Please read the React Hooks doc.